Feat/cap level x reserve price

0xDegenDeveloper opened the pull request:
- Fossil sends `cap_level` on chain instead of `volatility`
- Added `alpha` & `k` to job request params
- Cap level uses `alpha` & `k`, reserve price uses `cap_level` & `k`
- Reserve price calculation updated to use 30d logic instead of 7d
- Compiled repeated util functions into util file

0xDegenDeveloper committed:
update volatility calculation

0xDegenDeveloper committed:
update return type to match other endpoints

0xDegenDeveloper committed:
comment un-needed code

0xDegenDeveloper committed:
fix linter

0xDegenDeveloper committed:
Update job request & result structs

- Replace volatility with cap_level

- Add alpha & k

0xDegenDeveloper committed:
Add cap level endpoint

0xDegenDeveloper committed:
pass cap level into calculate_reserve_price

0xDegenDeveloper committed:
update cap level to return decimal f64

0xDegenDeveloper committed:
Update get_pricing_data.rs

Pass cap level result into calculate_reserve_price

0xDegenDeveloper committed:
update typing for on-chain

0xDegenDeveloper committed:
Add alpha and k to job request

0xDegenDeveloper committed:
update alpha and k types

0xDegenDeveloper committed:
clippy fix

0xDegenDeveloper committed:
update comment for clarity

0xDegenDeveloper committed:
Merge branch 'feat/update-vol-calculations' into feat/cap-level-X-reserve-price

0xDegenDeveloper committed:
Update grouping of data for volatility

0xDegenDeveloper committed:
linter fix

0xDegenDeveloper committed:
adding logging for cap level

0xDegenDeveloper committed:
update texting for volatility -> cap level

0xDegenDeveloper committed:
Move volatility function as helper in cap level endpoint

0xDegenDeveloper committed:
replace twap with DF logic

0xDegenDeveloper committed:
clean up pricing data request

ametel01 reviewed (approved):
PR is solid, we can marge after the merge conflicts are solved

0xDegenDeveloper committed:
add pricing util functions

- prepare data frame
- shared: `replace_timestamp_with_date` & `group_by_1h_interfaces`

0xDegenDeveloper committed:
update comments & use util functions

0xDegenDeveloper committed:
pass k into reserve price calculation

0xDegenDeveloper committed:
update reserve price calculation

- implement hourly grouping
- hard code window size
- update 7d logic to 30d
- use shared util functions

0xDegenDeveloper committed:
update comments

0xDegenDeveloper committed:
fix twap calculation

0xDegenDeveloper committed:
update location of `add_twaps`

- update comments for util functions
- remove old add_twap_7d function

0xDegenDeveloper committed:
cargo clippy fixes

0xDegenDeveloper committed:
Merge branch 'main' into feat/cap-level-X-reserve-price

0xDegenDeveloper committed:
Update README.md

0xDegenDeveloper committed:
Update README.md
